Add keyboard shortcut to refresh feed

TheDevJoao opened the pull request:
This PR resolves [#23660 ](https://github.com/mastodon/mastodon/issues/23660)

My solution adds a method called `handleHotkeyRefreshFeed` to the UI's `index` file, adding a keyboard shortcut  to the "." key. This method allows the user to scroll to the first article element in the page to better the overall experience of mouseless navigation.
 
#### Preview:
![scrollToTop](https://user-images.githubusercontent.com/64037198/234405865-dab3ccf4-85e4-4010-8d1b-1f2199c862dd.gif)

#### Updated legend:
![image](https://user-images.githubusercontent.com/64037198/234695049-3dd403a4-9303-4695-9285-29dc95a9beaa.png)

P.S: With my latest refactor the user can also scroll back to the top of the first toot/status of their home timeline in the advanced web interface's multi column mode but I'm not sure how to replicate this in other timelines: 
![advancedScrollToTop](https://user-images.githubusercontent.com/64037198/236552271-16aae098-3c87-4ed7-b924-521f7250b776.gif)

If anyone has a suggestion on how I can achieve this please let me know!

TheDevJoao committed:
add keyboard shortcut to scroll feed

- adds a high order component "refreshFeed"
- adds the hoc to the home, local and federated feeds

TheDevJoao committed:
adds the hotkey to keyboard shortcut legend

- adds the keyboard shortcut to the en, pt-PT and pt-BR locales.

TheDevJoao committed:
adds jest test
- tests the refreshFeed function

tribela commented:
`window.scroll` would be a problem in multi column mode.
Would be nice to look number shortcut (that selects the column and go to top of that column)

TheDevJoao committed:
refactor: changes test directory path and name

- refactors the test case with the corresponding function

TheDevJoao committed:
refactor: removes HOC, adds shortcut to keyMap

- Removed the HOC file and its association from the timeline files
- Refactored the scrollTo logic to find the timeline's first article
- Adds refreshFeed to the ui handler and keyMap

TheDevJoao commented:
> `window.scroll` would be a problem in multi column mode. Would be nice to look number shortcut (that selects the column and go to top of that column)

Hey @tribela! Thanks for pointing that out, I had  completely forgotten about multi columns in Mastodon 😆 
I recently refactored my code and partially solved this, check it out if you can! 🤝

TheDevJoao committed:
adds refresh timeline to defaultMessages.json

TheDevJoao committed:
adds translations from check-i18n.yml

TheDevJoao commented:
Hey @ClearlyClaire could you review this whenever you have time? 😄

TheDevJoao committed:
fix: ESlint enforcement

ClearlyClaire commented on app/javascript/mastodon/locales/defaultMessages.json:
```diff
@@ -3236,7 +3240,7 @@
         "id": "onboarding.steps.follow_people.title"
       },
       {
-        "defaultMessage": "You curate your own feed. Let's fill it with interesting people.",
+        "defaultMessage": "You curate your own feed. Lets fill it with interesting people.",
```
This has wrongfully snuck in during a rebase and should be fixed!

TheDevJoao replied:
> 
Hey, @ClearlyClaire, thanks for the review!

I used "refresh feed/timeline" in most of the implementation due to the issue's request and saw that a little blue bird also calls this feature in a similar fashion, but after opening the PR and  @tribela's comment, I figured the best solution would be to focus on the first element in a `scrollable` section.

After analyzing the codebase more thoroughly, I realized that there is already an implementation that does this called `focusColumn` that utilizes the `'1', '2', '3', '4', '5', '6', '7', '8', '9'` keys, but this is only true if the advanced web interface option is enabled.

Maybe my solution would make more sense if `refresh timeline` and `refresh feed` were called something else. Any suggestions on potential names to refactor?

ClearlyClaire reviewed (changes requested):
Hi! Thank you for your contribution!

I'm a bit confused, the description says it refreshes the feed, but what it does is focus the first post in the timeline? What am I missing here?

TheDevJoao commented on app/javascript/mastodon/features/ui/index.jsx:
```diff
@@ -485,6 +486,24 @@ class UI extends PureComponent {
   };
 
+  handleHotkeyRefreshFeed = () => {
+    const handleKeyDown = (event) => {
+      if (event.key === '.' && !event.ctrlKey && !event.shiftKey && !event.altKey && !event.metaKey) {
+        event.preventDefault();
+        const firstArticle = document.querySelector('article');
+        if (firstArticle) {
+          firstArticle.scrollIntoView();
+        }
+      }
+    };
+
+    window.addEventListener('keydown', handleKeyDown);
+    return () => {
+      window.removeEventListener('keydown', handleKeyDown);
+    };
```
> I don't understand why you are using this pattern of registering an event handler on the fly: managing the event handler is `react-hotkeys`'s job.

You're right, I'll make sure to fix this in my next commit.

TheDevJoao committed:
refactor: removes unnecessary event listeners

- also fixes a defaultMessage that was wrongfully rebased.

TheDevJoao committed:
fix: runs check-i18n to fix internationalization

TheDevJoao committed:
fix: removes new line from defaultMessages.json

mjankowski commented:
Closing due to lack of activity and merge conflicts -- that said, if you want to push this forward, can you rebase, and reopen, and then either a) update the description/names to reflect what the code actually does, or b) update the code to actually do the refresh that the desc/labels claim? (this seems more elaborate than what's there now, unless I'm missing something?)
